ami/ml/orchestration/nats_queue.py

```
@@ -330,9 +333,135 @@ async def delete_stream(self, job_id: int) -> bool:
+        Pulls from persistent advisory stream to find failed messages, then looks up task IDs.
+        Uses a durable consumer so acknowledged advisories are not re-delivered on subsequent calls.
+
+        Args:
+            job_id: The job ID (integer primary key)
+            n: Maximum number of task IDs to return (default: 10)
+
+        Returns:
+            List of task IDs that failed to process after max retry attempts
+        """
+        if self.nc is None or self.js is None:
+            raise RuntimeError("Connection is not open. Use TaskQueueManager as an async context manager.")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)